action/protocol/staking/handlers.go

```diff
@@ -178,16 +181,22 @@ func (p *Protocol) handleUnstake(ctx context.Context, act *action.Unstake, csm C
 	if err := csm.updateBucket(act.BucketIndex(), bucket); err != nil {
 		return log, errors.Wrapf(err, "failed to update bucket for voter %s", bucket.Owner.String())
 	}
-
-	weightedVote := p.calculateVoteWeight(bucket, csm.ContainsSelfStakingBucket(act.BucketIndex()))
+	selfStake, err := isSelfStakeBucket(featureCtx, csm, bucket.Index)
```

Comment:
move to L199, do not change the order of calls

Reply:
it's required by `calculateVoteWeight`